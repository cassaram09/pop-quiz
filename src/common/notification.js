import React from 'react'
import PropTypes from 'prop-types'

const Notification = ({ accept, cancel, message }) => {
  return (
    <div className="notification is-warning">
      <button className="delete" onClick={cancel.fn}></button>
      <p>{message}</p>
      <button onClick={accept.fn}>{accept.text}</button>
      <button onClick={cancel.fn} >{cancel.text}</button>
    </div>
  );
}

export default Notification;