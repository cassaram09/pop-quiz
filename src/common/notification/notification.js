import React from 'react'
import PropTypes from 'prop-types'

const Notification = ({ accept, cancel, message }) => {
  return (
    <div className="notification is-warning">
      <button className="delete notification__delete" onClick={cancel.fn}></button>
      <p className="notification__message">{message}</p>
      <button className="notification__accept" onClick={accept.fn}>{accept.text}</button>
      <button className="notification__cancel" onClick={cancel.fn} >{cancel.text}</button>
    </div>
  );
}

Notification.propTypes = {
  accept: PropTypes.shape({
    text: PropTypes.string,
    fn: PropTypes.func
  }),
  cancel: PropTypes.shape({
    text: PropTypes.string,
    fn: PropTypes.func
  }),
  message: PropTypes.string
};

Notification.defaultProps = {
  accept: {},
  cancel: {},
  message: ''
};

export default Notification;
