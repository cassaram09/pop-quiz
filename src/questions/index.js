export { default as Questions } from './questions.js'
export { default as Question } from './question.js'