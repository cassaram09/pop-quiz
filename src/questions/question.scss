.question {
  background: #eee;
  margin: 20px;
}

.question__title {
  font-weight: 700;
}