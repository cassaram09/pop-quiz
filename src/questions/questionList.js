import React, { Component } from 'react';

// import './questions.scss';

import { Question } from './'

class QuestionList extends Component {
  renderQuestions = data => data.map(d => <Question question={d} />)

  render() {
    return (
      <div className="questions">
        {this.renderQuestions(this.props.list)}
      </div>
    );
  }
}

export default QuestionList;