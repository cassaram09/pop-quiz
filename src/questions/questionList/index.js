export { default as QuestionList } from './questionList.js'