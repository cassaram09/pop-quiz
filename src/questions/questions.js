import React, { Component } from 'react';

// import './questions.scss';

import { Question } from './'

class Questions extends Component {
  renderQuestions = data => data.map(d => <Question question={d} />)

  render() {
    return (
      <div className="questions">
        {this.renderQuestions(this.props.questionList)}
      </div>
    );
  }
}

export default Questions;