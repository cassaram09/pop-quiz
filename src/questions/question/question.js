import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Notification } from 'common/notification'

class Question extends Component {
  state = {
    answerVisible: false,
    confirmationVisible: false,
  }

  // called when user wants to reveal answer
  confirm = () => {
    this.setState({ confirmationVisible: true })
  }

  // renders notification if user wants to reveal answer
  renderNotification = () => {
    if (this.state.confirmationVisible) {
      return( 
        <Notification
          accept={{ 
            text: "Show Answer", 
            fn: () => this.setState({ confirmationVisible: false, answerVisible: true }) 
          }}
          cancel={{ 
            text: "I'm not ready", 
            fn: () => this.setState({ confirmationVisible: false })
          }}
          message={"Are you sure you want to reveal the answer?"}
        />
      )
    } 
  }

  // renders answer if answer is revealed
  renderAnswer = () => {
    if (this.state.answerVisible){
      return <p className="question__answer">{this.props.answer}</p>
    }
  }

  // renders the button to allow for revealing of answer
  renderRevealButton = () => {
    if (!this.state.answerVisible) {
      return (
        <button 
          onClick={this.confirm} 
          className="button is-info question__reveal-answer"
        >
          Reveal Answer
        </button>
      )
    }
  }

  renderQuestion = () => {
    return <p className="question__text">{this.props.question}</p>
  }

  render() {
    return (
      <section className="section question">
        {this.renderQuestion()}
        {this.renderRevealButton()}
        {this.renderAnswer()}
        {this.renderNotification()}
      </section>
    );
  }
}

Question.propTypes = {
  answer: PropTypes.string,
  question: PropTypes.string,
  type: PropTypes.string
};

Question.defaultProps = {
  answer: '',
  question: '',
  type: ''
};

export default Question;
