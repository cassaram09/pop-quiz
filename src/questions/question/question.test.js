import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'utils/enzyme';

import { Question } from 'questions/question';
import { Notification } from 'common/notification'

describe('Question', () => {
  it('renders without crashing', () => {
    const question = shallow(<Question />);
  
    expect(question).toMatchSnapshot();
  });

  it('can handle null props', () => {
    const question = mount(<Question/>);

    expect(question.props().question).toEqual('');
    expect(question.props().answer).toEqual('');
    expect(question.props().type).toEqual('');
  });

  it('renders the question text', () => {
    const question = render(<Question question={"How are you?"}/>);

    expect(question.find('.question__text').length).toEqual(1)
    expect(question.find('.question__text').text()).toBe("How are you?");
  });

  it('renders the reveal button when answer is not revealed', () => {
    const question = render(<Question />);

    expect(question.find('.question__reveal-answer').length).toEqual(1)
  });

  it('does not render the reveal button when answer is revealed', () => {
    const question = mount(<Question />);

    question.setState({ answerVisible: true})

    expect(question.find('.question__reveal-answer').length).toEqual(0)
  });

  it('renders a notification when "Reveal Answer" is clicked', () => {
    const question = mount(<Question />);

    question.find('.question__reveal-answer').simulate('click')

    expect(question.find('Notification').length).toEqual(1)
  });
});
