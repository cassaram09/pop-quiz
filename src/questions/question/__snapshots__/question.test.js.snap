// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    answer=""
    question=""
    type=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="question__text"
        >
          
        </p>,
        <button
          className="button is-info question__reveal-answer"
          onClick={[Function]}
        >
          Reveal Answer
        </button>,
        undefined,
        undefined,
      ],
      "className": "section question",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "question__text",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reveal Answer",
          "className": "button is-info question__reveal-answer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reveal Answer",
        "type": "button",
      },
      undefined,
      undefined,
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="question__text"
          >
            
          </p>,
          <button
            className="button is-info question__reveal-answer"
            onClick={[Function]}
          >
            Reveal Answer
          </button>,
          undefined,
          undefined,
        ],
        "className": "section question",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "question__text",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reveal Answer",
            "className": "button is-info question__reveal-answer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reveal Answer",
          "type": "button",
        },
        undefined,
        undefined,
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
